### PR TITLE
Users auth and confirm

### PR DESCRIPTION
**Pull request by agarwalrashika00**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -18,6 +20,8 @@ gem 'importmap-rails', '~> 1.1.6'
 # Build JSON APIs with ease [https://github.com/rails/jbuilder]
 gem 'jbuilder', '~> 2.11.5'
 
+gem 'mailcatcher', '~> 0.2.4'
```
Comment (Ankita9818) [inline Gemfile:23]: Add letter opener in developent env only

File: README.md [hunk 2]
```diff
@@ -33,4 +33,4 @@ git clone https://github.com/agarwalrashika00/quizathon.git
 bundle exec rails server
 
 ## To visit user
```
Comment (Ankita9818) [inline README.md:35]: Update Readme

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -13,3 +13,21 @@
...
  *= require_self
  */
+ * {
+  box-sizing: border-box;
+ }
+ body {
+  margin-left: 10px;
+  margin-right: 10px;
+  background: #83a4d4;  /* fallback for old browsers */
+background: -webkit-linear-gradient(to right, #b6fbff, #83a4d4);  /* Chrome 10-25, Safari 5.1-6 */
```
Comment (Ankita9818) [inline app/assets/stylesheets/application.scss:23]: FIx indentation

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -13,3 +13,21 @@
...
+background: linear-gradient(to right, #b6fbff, #83a4d4); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
+
+}
+.field {
+  margin-top: 1.3rem;
+}
+.card {
+  margin-top: 2rem;
+  box-shadow: rgba(50, 50, 93, 0.25) 0px 50px 100px -20px, rgba(0, 0, 0, 0.3) 0px 30px 60px -30px, rgba(10, 37, 64, 0.35) 0px -2px 6px 0px inset;
+}
```
Comment (Ankita9818) [inline app/assets/stylesheets/application.scss:33]: Add new line

File: app/controllers/admin_controller.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+  def index
+  end
+
+  private def verify_if_admin
+    unless current_user.present? && current_user.role == 'admin'
+      redirect_to root_path, alert: "You don't have privileges to access this section"
+    end
+  end
+end
```
Comment (Ankita9818) [inline app/controllers/admin_controller.rb:12]: New line

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,2 +1,10 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
```
Comment (Ankita9818) [inline app/controllers/application_controller.rb:6]: Add a new line

File: app/controllers/users/confirmations_controller.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  # protected
+
+  # The path used after resending confirmation instructions.
+  # def after_resending_confirmation_instructions_path_for(resource_name)
+  #   super(resource_name)
+  # end
+
+  # The path used after confirmation.
+  def after_confirmation_path_for(resource_name, resource)
+    WelcomeUserMailer.with(user: resource).welcome_user_mail.deliver_later
```
Comment (Ankita9818) [inline app/controllers/users/confirmations_controller.rb:28]: Move this to model

File: app/views/users/_login_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<center>
+  <div class="card" style="width: 20rem;">
+    <h2 class="card-header">Log in</h2>
+    <% controller.class == Admin::SessionsController ? url = admin_sign_in_path : url = user_session_path %>
+    <%= form_for(resource, as: resource_name, url: url) do |f| %>
```
Comment (Ankita9818) [inline app/views/users/_login_form.html.erb:5]: Lets move to form_with

File: app/views/users/passwords/edit.html.erb [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<h2>Change your password</h2>
```
Comment (Ankita9818) [inline app/views/users/passwords/edit.html.erb:1]: Add i18n

File: config/initializers/constants.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+EMAIL_REGEXP = /\A[^@\s]+@[^@\s]+\z/
```
Comment (Ankita9818) [inline config/initializers/constants.rb:1]: Add these in module

File: app/views/users/_login_form.html.erb (hunk 8 above)
Comment (agarwalrashika00) [reply app/views/users/_login_form.html.erb:5]: https://trello.com/c/QUYxx5il/13-formwith-instead-of-formfor-in-devise-views

File: app/views/users/passwords/edit.html.erb (hunk 9 above)
Comment (agarwalrashika00) [reply app/views/users/passwords/edit.html.erb:1]: https://trello.com/c/f53Ex1gm/14-internationalization

File: Gemfile [hunk 11]
```diff
@@ -18,6 +22,8 @@ gem 'importmap-rails', '~> 1.1.6'
 # Build JSON APIs with ease [https://github.com/rails/jbuilder]
 gem 'jbuilder', '~> 2.11.5'
 
+gem "letter_opener", :group => :development
```
Comment (Ankita9818) [inline Gemfile:25]: group: :development

File: app/controllers/admin/base_controller.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+class Admin::BaseController < ApplicationController
+  before_action :ensure_is_admin
+
+  private def ensure_is_admin
+    unless current_user.present? && current_user.admin?
+      redirect_to root_path, alert: 'You donot have privileges to access this section'
+    end
+  end
+end
```
Comment (Ankita9818) [inline app/controllers/admin/base_controller.rb:9]: new line

**Pull request merged**